### PR TITLE
Benchmark subcommand to distinguish between DataFusion and Ballista

### LINKED ISSUE #401: TPC-H benchmark should distinguish between DataFusion and Ballista arguments
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
We can run benchmarks against DataFusion and Ballista from the same tpch binary. Many of the CLI arguments apply to both DataFusion and Ballista, but some are specific. For example `--concurrency` is only used when running with DataFusion, and `--host` and `--port` are only used when running against Ballista.

**Describe the solution you'd like**
There should be validation that we aren't providing parameters that make no sense for the context.  We could also consider adding subcommands for DataFusion vs Ballista.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by jgoday**
# Which issue does this PR close?

Closes #401.

 # Rationale for this change

# What changes are included in this PR?

It defines a BenchmarkSubCommandOpt to allow different arguments depending on whether it is run against ballista or datafusion.

# Are there any user-facing changes?

It changes tpch bin arugments generated by benchmarks project.
It contains changes in benchmarks/README.md and benchmars/run.sh files.

### CODE REVIEW

File: benchmarks/src/bin/tpch.rs [hunk 1]
```diff
@@ -100,6 +96,45 @@ struct BenchmarkOpt {
     port: Option<u16>,
 }
 
+#[derive(Debug, StructOpt, Clone)]
+struct DatafusionBenchmarkOpt {
```
Comment (andygrove) [inline benchmarks/src/bin/tpch.rs:100]: nit: Can we make this `DataFusionBenchmarkOpt`?

**Comment by andygrove** [comment]
There are two other files which will need updating:

- docs/user-guide/src/distributed/raspberrypi.md
- benchmarks/run.sh

**Comment by jgoday** [comment]
> There are two other files which will need updating:
> 
>     * docs/user-guide/src/distributed/raspberrypi.md
> 
>     * benchmarks/run.sh
Fixed DataFusionBenchmarkOpts name and raspberrypi.md in last commit.

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @jgoday

**Pull request merged**